### DISCUSSION TITLE
How to create Unique constraints and indexes from code?

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#2424] [Q&A] [answered]
How to create Unique constraints and indexes from code when creating collection?

**Comment by ganigeorgiev** [comment, marked as answer]
You can specify the collection indexes programmatically by updating the `Indexes` field. It is a slice of raw SQL `CREATE INDEX` expressions. For example:

```go
collection := &models.Collection{
	Name:    "example",
	Type:    models.CollectionTypeBase,
	Indexes: types.JsonArray[string]{
		"CREATE INDEX `your_unique_index_name1` ON `example` (`yourField`)",
		"CREATE UNIQUE INDEX `your_unique_index_name2` ON `example` (`yourField1`, `yourField2`)",
	},
	...
}

if err := app.Dao().SaveCollection(collection); err != nil {
    return err
}
```
_(the same apply if you use the [`forms.CollectionsUpsert`](https://pocketbase.io/docs/collection-methods/#create-new-collection-with-data-validations) when you need data validations)_

**Reply by bodaay** [reply]
please update the official documentation to reflect this, in the Advanced - collections, its still using 'Unique' in the examples provided
